nikola/plugins/compile/html.py

```diff
@@ -41,6 +41,10 @@ class CompileHtml(PageCompiler):
     name = "html"
     friendly_name = "HTML"
 
+    def compile_string(self, data, source_path=None, is_two_file=True, post=None, lang=None):
+        """Compile HTML into HTML strings, with shortcode support."""
+        return self.site.apply_shortcodes(data, with_dependencies=True, extra_context={'post': post})
```

Comment:
The `if not is_two_file:` part from `compile` should be moved here, too, shouldn't it?